search-service/src/test/kotlin/com/egm/stellio/search/authorization/permission/web/PermissionHandlerTests.kt

```diff
@@ -0,0 +1,529 @@
+import com.egm.stellio.shared.util.AuthContextModel.AUTH_PERMISSION_TERM
+import com.egm.stellio.shared.util.BEEHIVE_TYPE
+import com.egm.stellio.shared.util.JSON_LD_MEDIA_TYPE
+import com.egm.stellio.shared.util.JsonUtils.deserializeAsMap
+import com.egm.stellio.shared.util.MOCK_USER_SUB
+import com.egm.stellio.shared.util.RESULTS_COUNT_HEADER
+import com.egm.stellio.shared.util.Sub
+import com.egm.stellio.shared.util.SubjectType
+import com.egm.stellio.shared.util.ngsiLdDateTime
+import com.egm.stellio.shared.util.sub
+import com.egm.stellio.shared.util.toUri
+import com.ninjasquad.springmockk.MockkBean
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.confirmVerified
+import io.r2dbc.postgresql.codec.Json
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest
+import org.springframework.core.io.ClassPathResource
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import org.springframework.security.test.web.reactive.server.SecurityMockServerConfigurers.csrf
+import org.springframework.security.test.web.reactive.server.SecurityMockServerConfigurers.mockJwt
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.web.reactive.server.WebTestClient
+import java.net.URI
+import java.time.ZonedDateTime
+
+@ActiveProfiles("test")
+@EnableConfigurationProperties(ApplicationProperties::class, SearchProperties::class)
+@AutoConfigureWebTestClient(timeout = "30000")
+@WebFluxTest(PermissionHandler::class)
+class PermissionHandlerTests {
+
+    @MockkBean
+    private lateinit var permissionService: PermissionService
+
+    @Autowired
+    private lateinit var webClient: WebTestClient
+
+    @MockkBean
+    private lateinit var subjectReferentialService: SubjectReferentialService
+
+    @MockkBean
+    private lateinit var entityQueryService: EntityQueryService
+
+    @MockkBean
+    private lateinit var authorizationService: AuthorizationService
+
+    private val permissionUri = "/ngsi-ld/v1/auth/permissions"
+    private val id = "urn:ngsi-ld:Permission:1".toUri()
+    private val jwt = mockJwt().jwt { it.subject(MOCK_USER_SUB) }
+    private val userUuid = "55e64faf-4bda-41cc-98b0-195874cefd29"
+
+    @BeforeAll
+    fun configureWebClientDefaults() {
+        webClient = webClient.mutate()
+            .apply(jwt)
+            .apply(csrf())
+            .defaultHeaders {
+                it.accept = listOf(JSON_LD_MEDIA_TYPE)
+                it.contentType = JSON_LD_MEDIA_TYPE
+            }
+            .build()
+    }
+
+    fun gimmeRawPermission(
```

Comment:
you could remove some "fixed" parameters (type, createdAt, modifiedAt) to go below the threshold